src/libcore/iter.rs

```diff
@@ -2423,7 +2425,11 @@ macro_rules! step_impl {
             #[allow(trivial_numeric_casts)]
             fn steps_between(start: &$t, end: &$t, by: &$t) -> Option<usize> {
                 if *start <= *end {
-                    Some(((*end - *start) / *by) as usize)
+                    // Note: We assume $t <= usize/isize here
+                    Some(
+                        ((*end as isize).wrapping_sub(*start as isize) as usize
```

Comment:
Since you ask, I should probably insert a comment there.

It's needed because the length of a Range<i8> or Range<i64> etc may be larger than the maximum of the signed type. We compute a wrapping sub and cast to unsigned and get the correct value.